### PR TITLE
Parallel Gap Filling for Same Range Task

### PR DESCRIPTION
**Pull request by mwang87**
Using Java Stream to handle parallel processing.

### CODE REVIEW

**Comment by robinschmid** [comment]
Hey Ming,
sounds great. I am going to test it today.

Cheers,
Robin

File: src/main/java/net/sf/mzmine/modules/peaklistmethods/gapfilling/samerange/SameRangeTask.java [hunk 1]
```diff
@@ -55,179 +59,246 @@
...
-	    // Canceled?
-	    if (isCanceled())
-		return;
+        /*************************************************************
+            Creating a stream to process the data in parallel
+        */
 
-	    PeakListRow sourceRow = peakList.getRow(row);
-	    PeakListRow newRow = new SimplePeakListRow(sourceRow.getID());
+        List<Integer> rowList = new ArrayList<Integer>();
```
Comment (robinschmid) [inline src/main/java/net/sf/mzmine/modules/peaklistmethods/gapfilling/samerange/SameRangeTask.java]: You could also directly create the parallel stream

`IntStream.range(0, totalRows).parallel().forEach`

File: src/main/java/net/sf/mzmine/modules/peaklistmethods/gapfilling/samerange/SameRangeTask.java [hunk 2]
```diff
@@ -55,179 +59,246 @@
...
+            }
+
+            processedPeakList.addRow(newRow);
+
+            processedRows++;
+
+        }*/
+
+        // Append processed peak list to the project
+        project.addPeakList(processedPeakList);
```
Comment (robinschmid) [inline src/main/java/net/sf/mzmine/modules/peaklistmethods/gapfilling/samerange/SameRangeTask.java]: Need to check for isCanceled - as streams are not able to

` if (isCanceled())  return;`

**Comment by mwang87** [comment]
Thank Robin, I've also removed some of the comments but overall it looks good. I think we'll wait for your first set of changes to go in, and then I'll pull from upstream and we'll test and get these changes in!

Should help a lot for more people to run gap filling quicker. 

Ming

**Comment by tomas-pluskal** [comment]
Looks good, guys. Are you still working on it or should I merge it now?

**Comment by tomas-pluskal** [comment]
Hi Robin,
I am waiting for @evgerher to fix some bugs in his new SIRIUS identification module. After that, I will prepare a new release. There will be a lot of updates included :)
I expect that to happen by mid-September.

**Pull request merged**